app/services/build_defect.rb

```diff
@@ -2,22 +2,38 @@ class BuildDefect
   attr_accessor :defect_params,
                 :property_id,
                 :communal_area_id,
-                :priority_id
+                :priority_id,
+                :created_at,
+                :defect
 
   def initialize(defect_params:, options: {})
     self.defect_params = defect_params
     self.property_id = options[:property_id]
     self.communal_area_id = options[:communal_area_id]
     self.priority_id = options[:priority_id]
+    self.created_at = options[:created_at]
   end
 
   def call
-    defect = Defect.new(defect_params)
+    self.defect = Defect.new(defect_params)
     defect.property = Property.find(property_id) if property_id.present?
     defect.communal_area = CommunalArea.find(communal_area_id) if communal_area_id.present?
     defect.priority = Priority.find(priority_id) if priority_id.present?
     defect.set_target_completion_date
 
+    set_created_at if created_at.present?
+
     defect
   end
+
+  private
+
+  def set_created_at
+    date_parts = created_at.values_at(:day, :month, :year)
+    return unless date_parts.all?(&:present?)
+
+    day, month, year = date_parts.map(&:to_i)
+    date = Date.new(year, month, day)
+    defect.set_created_at(date)
+  end
```

Comment:
This is here and in `EditDefect`. Is there any way we could dedupe this functionality?

Reply:
Great spot, I've pulled this out into a parent class that both inherit from. This follows a pattern already established in the app (versus including a module).